src/Jobs/MemberOrchestrator.php

```diff
@@ -123,13 +124,20 @@ private function processMappingBase()
         $nickname = null;
         $pending_drops = collect();
         $pending_adds = collect();
+        $nickname_diff = false;
 
         $discord_user = DiscordUser::where('discord_id', $this->member->user->id)->first();
 
         if (! is_null($discord_user)) {
-            if (! is_null($discord_user->group->main_character))
-                $nickname = $discord_user->group->main_character->name;
-
+            if (! is_null($discord_user->group->main_character)) {
+                $corp_id = $discord_user->group->main_character->corporation()->corporation_id;
```

Comment:
This can be simplified a bit.

`corporation` is a relation link to `CorporationInfo` - so if you're using it, you can directly set `$corp = $discord_user->group->main_character->corporation;`.

In all case, you have to ensure you got a `CorporationInfo` instance, otherwise you'll get exception later in the code.